File: postman/sdks/Specs SDKs/Search Service/typescript/src/services/search-endpoints/models/artists-response.ts
import { z } from 'zod';
import {
  ArtistsDefaultModel,
  artistsDefaultModel,
  artistsDefaultModelRequest,
  artistsDefaultModelResponse,
} from './artists-default-model';

/**
 * Zod schema for the ArtistsResponse model.
 * Defines the structure and validation rules for this data type.
 * This is the shape used in application code - what developers interact with.
 */
export const artistsResponse = z.lazy(() => {
  return z.object({
    debug: z.any().optional().nullable(),
    explain: z.any().optional().nullable(),
    data: z.array(artistsDefaultModel),
  });
});

/**
 *
 * @typedef  {ArtistsResponse} artistsResponse
 * @property {any}
 * @property {any}
 * @property {ArtistsDefaultModel[]} - List of artist models.
 */
export type ArtistsResponse = z.infer<typeof artistsResponse>;

/**
 * Zod schema for mapping API responses to the ArtistsResponse application shape.
 * Handles any property name transformations from the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const artistsResponseResponse = z.lazy(() => {
  return z
    .object({
      debug: z.any().optional().nullable(),
      explain: z.any().optional().nullable(),
      data: z.array(artistsDefaultModelResponse),
    })
    .transform((data) => ({
      debug: data['debug'],
      explain: data['explain'],
      data: data['data'],
    }));
});

/**
 * Zod schema for mapping the ArtistsResponse application shape to API requests.
 * Handles any property name transformations required by the API schema.
 * If property names match the API schema exactly, this is identical to the application shape.
 */
export const artistsResponseRequest = z.lazy(() => {
  return z
    .object({
      debug: z.any().optional().nullable(),
      explain: z.any().optional().nullable(),
      data: z.array(artistsDefaultModelRequest),
    })
    .transform((data) => ({
      debug: data['debug'],
      explain: data['explain'],
      data: data['data'],
    }));
});

File: postman/sdks/Specs SDKs/Search Service/python/src/search_service_sdk/models/artists_response.py
from __future__ import annotations
from typing import List
from pydantic import Field
from typing import Optional
from typing import Union
from .utils.base_model import BaseModel
from .artists_default_model import ArtistsDefaultModel


class ArtistsResponse(BaseModel):
    """ArtistsResponse

    :param debug: debug, defaults to None
    :type debug: dict, optional
    :param explain: explain, defaults to None
    :type explain: dict, optional
    :param data: List of artist models.
    :type data: List[ArtistsDefaultModel]
    """

    debug: Optional[dict] = Field(default=None)
    explain: Optional[dict] = Field(default=None)
    data: List[ArtistsDefaultModel] = Field(description="List of artist models.")
